hal/src/generate/generate-java-resources.py

```diff
@@ -0,0 +1,64 @@
+
+import os
+import sys
```

Comment:
The existing version wasn't (in its current state) reusable here. They could probably be refactored and combined.